### PR TITLE
Fix TypeScript Errors and add Redux

### LINKED ISSUE #54: PWA: Make 1 call to the spreadsheet instead of scattered calls
Right now there are separate calls for the google spreadsheet. In order to make the app easier to migrate to a PWA, we need to make 1 or a few calls to the google spreadsheet instead of calls scattered around.

For example, the search page pulls the google sheet and the home page does as well. These can both share the google sheet state (which should be similar to a commander-style design pattern).

### Results

The google spreadsheet should be **a single source of truth**, or in other words, there should be **exclusively 1 place where the sheet state exists** and should only be pulled if necessary.

### PR DESCRIPTION
**Pull request by Alex-Cannon**
### Redux Refactoring
Close #54 - WIP, but TypeScript & redux were added and have basic setup
* I pushed the wrong branch for my previous PR. This is the correct branch.

- [X] Installed `redux react-redux redux-thunk`
- [X] Initialized Redux store
- [X] Converted ./src/index.jsx to .tsx and updated `/src/index.html`
- [ ] Refactored `useAPI` with redux
- [ ] Refactored `/categories` and `/search` pages to

### Fixed TypeScript Path Error
* When I use "\~/core/..." or any "\~" paths, TypeScript throws the error: "Cannot find module".

- [X] Fixed TypeScript and Parcel issues via [official parcel docs](https://parceljs.org/typeScript.html)
- [X] Installed `@types/node` to allow `process` to be typed globally.

My solution: Use `tsconfig.json`, `package.json` and the official parcel docs to fix this.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,14 +49,18 @@
 	},
 	"dependencies": {
 		"@types/jest": "^24.0.23",
+		"@types/node": "^12.12.11",
```
Comment (Vpr99) [inline package.json]: I'm curious, what do we need Node types for?

File: src/core/store/services/useServices.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * 1. GET all google sheets
+ * 2. Store globally in array
+ * 3. Hooks pull their data from the array
+ */
+
+// Returns a list of categories
+export const useCategories = () => {};
+
+// Returns a list of subtypes for a category
+//export const useSubTypes = category => {};
```
Comment (Vpr99) [inline src/core/store/services/useServices.ts]: Are these being used? If not could we get rid of them for now?

File: package.json (hunk 1 above)
Comment (Alex-Cannon) [reply package.json]: I think this was an issue to do with making TypeScript recognize `process.env` as a global. I should look into it as it's been a few weeks.

File: src/core/store/services/useServices.ts (hunk 2 above)
Comment (Alex-Cannon) [reply src/core/store/services/useServices.ts]: Yeah for sure. I think for now, I should remove any code like this or that isn't being used. The app doesn't use redux yet, but I think it'd be good to merge in TypeScript for now.

**Comment by Alex-Cannon** [comment]
Because of git wierdness, I'm creating a new PR for this.

**Pull request closed without merging**